src/main/java/io/supertokens/webserver/api/userroles/AddUserRoleAPI.java

```diff
@@ -48,7 +48,7 @@ public String getPath() {
 
     @Override
     protected void doPut(HttpServletRequest req, HttpServletResponse resp) throws IOException, ServletException {
-        // API is tenant specific
+        // API is tenant specific, but uses public tenant storage
         JsonObject input = InputParser.parseJsonObjectOrThrowError(req);
         String userId = InputParser.parseStringOrThrowError(input, "userId", false);
```

Comment:
we will add the roles and user id mapping to the target storage only. We will remove the FK constraint from the user <-> role and role tables, and when deleting a role, we will loop through all storages first and delete the roles from there, and then from the public storage.